### PR TITLE
Task execution workflow refactoring

### PR DESCRIPTION
**Pull request by whiterabbit1983**
### **PR Type**
Enhancement, Tests

___

### **Description**
- Refactored task execution workflow to modularize step handling.
  - Introduced individual handler methods for each step type.
  - Replaced monolithic step handling logic with a dynamic dispatch mechanism.

- Added new `execute_api_call` activity for API call execution.

- Fixed typo in module name and updated imports accordingly.

- Added extensive unit tests for task execution workflow.
  - Covered various step types and scenarios, including error handling.

___

### **Changes walkthrough** 📝
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Enhancement</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>execute_api_call.py</strong><dd><code>Introduced `execute_api_call` function for API calls</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

agents-api/agents_api/activities/execute_api_call.py

<li>Added <code>execute_api_call</code> function to handle API call execution.<br> <li> Defined <code>RequestArgs</code> TypedDict for request parameters.<br> <li> Implemented error handling and response parsing logic.<br> <li> Integrated mock execution for testing environments.

</details>

  </td>
  <td><a href="https://github.com/julep-ai/julep/pull/1078/files#diff-bbd41da2245c7eb5f9888933d9205d067d35df6496f6fc3581fb28619b7a1b52">[link]</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>__init__.py</strong><dd><code>Refactored task execution workflow step handling</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

agents-api/agents_api/workflows/task_execution/__init__.py

<li>Refactored step handling logic into individual handler methods.<br> <li> Replaced monolithic step handling with dynamic dispatch.<br> <li> Fixed typo in module import (<code>excecute_api_call</code> to <code>execute_api_call</code>).<br> <li> Added support for various step types like <code>LogStep</code>, <code>SwitchStep</code>, etc.

</details>

  </td>
  <td><a href="https://github.com/julep-ai/julep/pull/1078/files#diff-d12de13df920a717fe2e94addd7e85fbb6216144a71cd1aba10e84f44cecf9f1">+474/-436</a></td>

</tr>
</table></td></tr><tr><td><strong>Bug fix</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>worker.py</strong><dd><code>Fixed typo in module import</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

agents-api/agents_api/worker/worker.py

<li>Fixed typo in module import (<code>excecute_api_call</code> to <code>execute_api_call</code>).

</details>

  </td>
  <td><a href="https://github.com/julep-ai/julep/pull/1078/files#diff-1f0b71d866cecdf7126078ef62a72a04bfccf7a03156f351df448c5f536c9c89">+1/-1</a>&nbsp; &nbsp; &nbsp; </td>

</tr>
</table></td></tr><tr><td><strong>Tests</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>test_task_execution_workflow.py</strong><dd><code>Added unit tests for task execution workflow</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

agents-api/tests/test_task_execution_workflow.py

<li>Added unit tests for task execution workflow.<br> <li> Covered various step types including <code>ToolCallStep</code>, <code>SwitchStep</code>, etc.<br> <li> Tested error scenarios and edge cases.

</details>

  </td>
  <td><a href="https://github.com/julep-ai/julep/pull/1078/files#diff-90d03c9dad92cb6056e99f07aa50711df681cefe722f051bdca0a129cb86b909">+478/-0</a>&nbsp; </td>

</tr>
</table></td></tr></tr></tbody></table>

___

> <details> <summary>  Need help?</summary><li>Type <code>/help how to ...</code> in the comments thread for any questions about Qodo Merge usage.</li><li>Check out the <a href="https://qodo-merge-docs.qodo.ai/usage-guide/">documentation</a> for more information.</li></details>

----

> [!IMPORTANT]
> Refactor task execution workflow for modularity, add `execute_api_call` function, fix import typo, and add unit tests.
> 
>   - **Refactoring**:
>     - Modularized task execution workflow in `__init__.py` by introducing individual handler methods for each step type and dynamic dispatch.
>     - Added `execute_api_call` function in `execute_api_call.py` for API call execution.
>   - **Bug Fix**:
>     - Fixed typo in module import from `excecute_api_call` to `execute_api_call` in `worker.py` and `__init__.py`.
>   - **Testing**:
>     - Added unit tests in `test_task_execution_workflow.py` covering various step types and scenarios, including error handling.
> 
> <sup>This description was created by </sup>[Ellipsis](https://www.ellipsis.dev?ref=julep-ai%2Fjulep&utm_source=github&utm_medium=referral)<sup> for d4a33110d5d6f83aea0a510df13587208e3a6753. It will automatically update as commits are pushed.</sup>

### CODE REVIEW

File: agents-api/agents_api/activities/task_steps/prompt_step.py [hunk 1]
```diff
@@ -78,15 +77,6 @@ async def prompt_step(context: StepContext) -> StepOutcome:
     # Wrap the prompt in a list if it is not already
     prompt = prompt if isinstance(prompt, list) else [{"role": "user", "content": prompt}]
```
Comment (creatorrr) [inline agents-api/agents_api/activities/task_steps/prompt_step.py:79]: this file's changes should not go into dev just yet. it'll break stuff. instead #1081 should do that later (same for log step activity)

please roll this file back @whiterabbit1983

Comment (whiterabbit1983) [reply agents-api/agents_api/activities/task_steps/prompt_step.py]: I reverted this
